doc/classes/Dictionary.xml

```diff
@@ -21,11 +21,20 @@
 		var points_dict = {"White": 50, "Yellow": 75, "Orange": 100}
 
 		# Alternative Lua-style syntax.
+		#
 		# Doesn't require quotes around keys, but only string constants can be used as key names.
-		# Additionally, key names must start with a letter or an underscore.
+		# Additionally, for key names to be usable without quotes, they must start with a letter
+		# or an underscore and must not be reserved identifiers (such as `class` or `while`).
+		# It is still possible to use reserved identifiers, or key names that start with a digit,
+		# but they must be quoted.
+		# Note that while you can mix dictionary syntaxes in a single file,
+		# you can't use both dictionary syntaxes in a single dictionary declaration.
```

Comment:
I'd personally rather move all of the LUA-style explanation out of this code snippet. It feels nasty to explain all of this inside a single `codeblock` tag.